gh-149085: Add max_threads keyword to faulthandler.dump_traceback() (#149106)
python/cpython: Doc/library/faulthandler.rst

* Each string is limited to 500 characters.
* Only the filename, the function name and the line number are
  displayed. (no source code)
* It is limited to 100 frames per thread, and by default to 100 threads

>   total in newest-first order (configurable via *max_threads*).

Reviewer: ```suggestion
* It is limited to 100 frames per thread, and 100 threads
  (configurable via *max_threads*).
```

Dumping the traceback
---------------------

.. function:: dump_traceback(file=sys.stderr, all_threads=True, *, max_threads=100)

   Dump the tracebacks of all threads into *file*. If *all_threads* is
   ``False``, dump only the current thread. *max_threads* caps the number

>    of threads dumped; a ``...`` marker is written if there are more.

Reviewer: > a ``...`` marker is written if there are more

That's an implementation detail, please don't document it (remove it from the doc). Same remark for all documentation of this PR.

   .. seealso:: :func:`traceback.print_tb`, which can be used to print a traceback object.

   .. versionchanged:: 3.5
      Added support for passing file descriptor to this function.

>    .. versionchanged:: 3.15

Reviewer: ```suggestion
   .. versionchanged:: next
```

   .. versionchanged:: 3.7
      This function is now always available.

>    .. versionchanged:: 3.15

Reviewer: ```suggestion
   .. versionchanged:: next
```
